### PR TITLE
Add inotify, rebuild, watching and debug instructions to the dev doc.

### PR DESCRIPTION
**Pull request by hexa00**
Signed-off-by: Antoine Tremblay <antoine.tremblay@ericsson.com>

### CODE REVIEW

File: doc/Developing.md [hunk 1]
```diff
@@ -62,6 +73,63 @@ It can also be started with:
...
+
+This means that the examples have their own version of theia-core in their
+node_modules directory.
+
+So each time you make a change in the theia-core you need to synchronize
+the theia-core in the example's node_modules directory.
+
+The current way to do so is to run this command in the example directory:
+
+	$ rm -rf node_modules/theia && npm install
```
Comment (kittaakos) [inline doc/Developing.md]: This command seems to be obsolete.

Comment (hexa00) [inline doc/Developing.md]: How so ? #125 makes the watch not working and the file-depencendy-updater is not in the electron example

Comment (kittaakos) [inline doc/Developing.md]: FYI: executing the `npm run watch` command on `theia-core` and then starting the browser example with `npm run start` includes the hot reload functionality. So if you change the client code in `theia-core` your browser get refreshed with the most recent content. It is unfortunate that it does not work correctly for you.

Comment (hexa00) [inline doc/Developing.md]: Humm the hot reload did not work for me ? Is it supposed to work with backend & frontend code ?

Comment (hexa00) [inline doc/Developing.md]: Note I just retried and there's nothing to be done :( the file-updater sees the change but no hot-reload, reloading the page doesn't help either... but we can discuss this as part of #125

**Comment by hexa00** [comment]
Thanks, I'll wait for Simon to take a look before merging

File: doc/Developing.md [hunk 2]
```diff
@@ -62,6 +73,63 @@ It can also be started with:
...
+
+Work is in progress make this step unnecessary or more automated.
```
Comment (simark) [inline doc/Developing.md]: missing "to"

Comment (hexa00) [inline doc/Developing.md]: yep thx

**Comment by simark** [comment]
I am trying out the watch system at the moment, so I'll take a closer look to this patch once I have a better understanding.

**Pull request merged**